host_pool_plugin/tasks.py

```diff
@@ -0,0 +1,65 @@
+# #######
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+import httplib
+import os
+import requests
+
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify.decorators import operation
+
+
+@operation
+def acquire(service_url, **kwargs):
+    ctx.logger.info("Acquire host")
+    response = requests.post('{0}/hosts'.format(service_url))
+    host = response.json()
+    ctx.logger.info("Response received: {0}".format(str(host)))
+    if response.status_code == httplib.CREATED:
+        ctx.instance.runtime_properties['ip'] = host['host']
+        key_content = host['auth'].get('keyfile')
+        agent = {'user': host['auth']['username'],
+                 'port': host['port']}
+        if key_content:
+            key_path = _save_keyfile(key_content, host['host_id'])
+            agent['key'] = key_path
+        else:
+            agent['password'] = host['auth'].get('password', '')
+        ctx.instance.runtime_properties['cloudify_agent'] = agent
```

Comment:
OK, I'lll change it. I just wanted to be consistent with the naming -> in properties cloudify_agent dictionary is for holding such information - that's why I used the same name in runtime properties.